src/pybind/mgr/cephadm/services/iscsi.py

```diff
@@ -26,6 +26,13 @@ def config(self, spec: IscsiServiceSpec) -> None:
 
     def prepare_create(self, daemon_spec: CephadmDaemonSpec[IscsiServiceSpec]) -> CephadmDaemonSpec:
         assert self.TYPE == daemon_spec.daemon_type
+        # if spec is not attached to daemon_spec it is likely a redeploy or reconfig and
+        # spec should be in spec store
```

Comment:
can we please fail hard, if we don't have a spec for iscsi?

Reply:
can you clarify what you mean by fail hard here (and in similar comments)? If we can't find the spec the `assert daemon_spec.spec` will fail a few lines later. Do you want to fail in a different way?